xonsh/imphooks.py

```diff
@@ -60,6 +64,16 @@ def find_spec(self, fullname, path, target=None):
     #
     # SourceLoader methods
     #
+    def create_module(self, spec):
+        """Create a xonsh module with the appropriate attributes."""
+        name = spec.name
+        parent = spec.parent
+        mod = XonshModule(name)
```

Comment:
Do we need a special module type? Can this line be `mod = types.ModuleType(name)`?